# Binary options put, call statistical research 

This Notebook analyse forex data from EURUSD pair M5 timeframe for the binary options market 
the data is from the duration of january 1st 2020 to december 31st 2020, the aim of this research is to find a pattern 
the placing trades in the binary options markert, in other to accurately predict where a market will close, from 
the previous candles, if it closes above the open price of the candle in question it is a call option, if it closes below it is a put option, when the open price and the close price is the same this is a no trade option which is statistically rare, so this won't be a focal point in this study

In [1]:
#import the relevant packages
import pandas as pd
import numpy as np
import seaborn as sns
sns.set(color_codes=True)

In [2]:
#read the data from a csv file
data = pd.read_csv("EURUSD_M5_202001020600_202012030000.csv", sep="\t")
data

<DATE>    <TIME>   <OPEN>   <HIGH>    <LOW>  <CLOSE>  <TICKVOL>  \
0      2020.01.02  06:00:00  1.12132  1.12134  1.12128  1.12132         61   
1      2020.01.02  06:05:00  1.12132  1.12134  1.12124  1.12133         55   
2      2020.01.02  06:10:00  1.12133  1.12142  1.12132  1.12142         56   
3      2020.01.02  06:15:00  1.12142  1.12143  1.12137  1.12142         87   
4      2020.01.02  06:20:00  1.12143  1.12143  1.12122  1.12126         68   
...           ...       ...      ...      ...      ...      ...        ...   
68938  2020.12.02  23:40:00  1.21166  1.21181  1.21147  1.21178        151   
68939  2020.12.02  23:45:00  1.21176  1.21177  1.21153  1.21166        154   
68940  2020.12.02  23:50:00  1.21168  1.21184  1.21164  1.21172        143   
68941  2020.12.02  23:55:00  1.21172  1.21175  1.21134  1.21136        158   
68942  2020.12.03  00:00:00  1.21137  1.21157  1.21102  1.21151         56   

       <VOL>  <SPREAD>  
0          0         3  
1          0         2  
2          0         2  
3          0         3  
4          0         3  
...      ...       ...  
68938      0         1  
68939      0         1  
68940      0         1  
68941      0         1  
68942      0         1  

[68943 rows x 9 columns]

In [3]:
#drop the <VOL>, column as it is not needed in this research
data.drop("<VOL>", inplace=True, axis=1)

In [4]:
data

<DATE>    <TIME>   <OPEN>   <HIGH>    <LOW>  <CLOSE>  <TICKVOL>  \
0      2020.01.02  06:00:00  1.12132  1.12134  1.12128  1.12132         61   
1      2020.01.02  06:05:00  1.12132  1.12134  1.12124  1.12133         55   
2      2020.01.02  06:10:00  1.12133  1.12142  1.12132  1.12142         56   
3      2020.01.02  06:15:00  1.12142  1.12143  1.12137  1.12142         87   
4      2020.01.02  06:20:00  1.12143  1.12143  1.12122  1.12126         68   
...           ...       ...      ...      ...      ...      ...        ...   
68938  2020.12.02  23:40:00  1.21166  1.21181  1.21147  1.21178        151   
68939  2020.12.02  23:45:00  1.21176  1.21177  1.21153  1.21166        154   
68940  2020.12.02  23:50:00  1.21168  1.21184  1.21164  1.21172        143   
68941  2020.12.02  23:55:00  1.21172  1.21175  1.21134  1.21136        158   
68942  2020.12.03  00:00:00  1.21137  1.21157  1.21102  1.21151         56   

       <SPREAD>  
0             3  
1             2  
2             2  
3             3  
4             3  
...         ...  
68938         1  
68939         1  
68940         1  
68941         1  
68942         1  

[68943 rows x 8 columns]

In [5]:
#set call when open price is less than close price
data["<UP>"] = data["<OPEN>"] < data["<CLOSE>"]

In [6]:
#set put when open price is greater than close price
data["<DOWN>"] = data["<OPEN>"] > data["<CLOSE>"]

In [7]:
#no trade when open price is equal to close price
data["<NO_MOVE>"] = data["<OPEN>"] == data["<CLOSE>"]

In [8]:
data

<DATE>    <TIME>   <OPEN>   <HIGH>    <LOW>  <CLOSE>  <TICKVOL>  \
0      2020.01.02  06:00:00  1.12132  1.12134  1.12128  1.12132         61   
1      2020.01.02  06:05:00  1.12132  1.12134  1.12124  1.12133         55   
2      2020.01.02  06:10:00  1.12133  1.12142  1.12132  1.12142         56   
3      2020.01.02  06:15:00  1.12142  1.12143  1.12137  1.12142         87   
4      2020.01.02  06:20:00  1.12143  1.12143  1.12122  1.12126         68   
...           ...       ...      ...      ...      ...      ...        ...   
68938  2020.12.02  23:40:00  1.21166  1.21181  1.21147  1.21178        151   
68939  2020.12.02  23:45:00  1.21176  1.21177  1.21153  1.21166        154   
68940  2020.12.02  23:50:00  1.21168  1.21184  1.21164  1.21172        143   
68941  2020.12.02  23:55:00  1.21172  1.21175  1.21134  1.21136        158   
68942  2020.12.03  00:00:00  1.21137  1.21157  1.21102  1.21151         56   

       <SPREAD>   <UP>  <DOWN>  <NO_MOVE>  
0             3  False   False       True  
1             2   True   False      False  
2             2   True   False      False  
3             3  False   False       True  
4             3  False    True      False  
...         ...    ...     ...        ...  
68938         1   True   False      False  
68939         1  False    True      False  
68940         1   True   False      False  
68941         1  False    True      False  
68942         1   True   False      False  

[68943 rows x 11 columns]

In [9]:
#convert these columns to type int so they appear as one hot varibles
data[["<UP>", "<DOWN>", "<NO_MOVE>"]] = data[["<UP>", "<DOWN>", "<NO_MOVE>"]].astype("int32")

In [10]:
data

<DATE>    <TIME>   <OPEN>   <HIGH>    <LOW>  <CLOSE>  <TICKVOL>  \
0      2020.01.02  06:00:00  1.12132  1.12134  1.12128  1.12132         61   
1      2020.01.02  06:05:00  1.12132  1.12134  1.12124  1.12133         55   
2      2020.01.02  06:10:00  1.12133  1.12142  1.12132  1.12142         56   
3      2020.01.02  06:15:00  1.12142  1.12143  1.12137  1.12142         87   
4      2020.01.02  06:20:00  1.12143  1.12143  1.12122  1.12126         68   
...           ...       ...      ...      ...      ...      ...        ...   
68938  2020.12.02  23:40:00  1.21166  1.21181  1.21147  1.21178        151   
68939  2020.12.02  23:45:00  1.21176  1.21177  1.21153  1.21166        154   
68940  2020.12.02  23:50:00  1.21168  1.21184  1.21164  1.21172        143   
68941  2020.12.02  23:55:00  1.21172  1.21175  1.21134  1.21136        158   
68942  2020.12.03  00:00:00  1.21137  1.21157  1.21102  1.21151         56   

       <SPREAD>  <UP>  <DOWN>  <NO_MOVE>  
0             3     0       0          1  
1             2     1       0          0  
2             2     1       0          0  
3             3     0       0          1  
4             3     0       1          0  
...         ...   ...     ...        ...  
68938         1     1       0          0  
68939         1     0       1          0  
68940         1     1       0          0  
68941         1     0       1          0  
68942         1     1       0          0  

[68943 rows x 11 columns]

In [11]:
open_close = (data["<OPEN>"] - data["<CLOSE>"]) * 100000
open_low = (data["<OPEN>"] - data["<LOW>"]) * 100000
open_high = (data["<OPEN>"] - data["<HIGH>"]) * 100000
close_low = (data["<CLOSE>"] - data["<LOW>"]) * 100000
close_high = (data["<CLOSE>"] - data["<HIGH>"]) * 100000
high_low = (data["<HIGH>"] - data["<LOW>"]) * 100000

In [12]:
#create a dataframe of the subtracted columns
move_df = pd.DataFrame({"open_close" : open_close, 
                        "open_low"   : open_low,
                        "open_high"  : open_high,
                        "close_low"  : close_low,
                        "close_high" : close_high,
                        "high_low"   : high_low, })
move_df

open_close  open_low  open_high  close_low  close_high  high_low
0             0.0       4.0       -2.0        4.0        -2.0       6.0
1            -1.0       8.0       -2.0        9.0        -1.0      10.0
2            -9.0       1.0       -9.0       10.0         0.0      10.0
3             0.0       5.0       -1.0        5.0        -1.0       6.0
4            17.0      21.0        0.0        4.0       -17.0      21.0
...           ...       ...        ...        ...         ...       ...
68938       -12.0      19.0      -15.0       31.0        -3.0      34.0
68939        10.0      23.0       -1.0       13.0       -11.0      24.0
68940        -4.0       4.0      -16.0        8.0       -12.0      20.0
68941        36.0      38.0       -3.0        2.0       -39.0      41.0
68942       -14.0      35.0      -20.0       49.0        -6.0      55.0

[68943 rows x 6 columns]

### create a future dataframe to store the next data i.e the candle in question  

In [13]:
future = data.iloc[1:,8:11]

In [14]:
future.columns = ["up*", "down*", "no_move*"]

In [15]:
future.reset_index(inplace=True)
future.drop(["index"], axis = 1, inplace=True)

In [16]:
#concat move_df and future to create a new dataframe
dataset = pd.concat([move_df, data.iloc[:,2:6], future], axis = 1)
dataset

open_close  open_low  open_high  close_low  close_high  high_low  \
0             0.0       4.0       -2.0        4.0        -2.0       6.0   
1            -1.0       8.0       -2.0        9.0        -1.0      10.0   
2            -9.0       1.0       -9.0       10.0         0.0      10.0   
3             0.0       5.0       -1.0        5.0        -1.0       6.0   
4            17.0      21.0        0.0        4.0       -17.0      21.0   
...           ...       ...        ...        ...         ...       ...   
68938       -12.0      19.0      -15.0       31.0        -3.0      34.0   
68939        10.0      23.0       -1.0       13.0       -11.0      24.0   
68940        -4.0       4.0      -16.0        8.0       -12.0      20.0   
68941        36.0      38.0       -3.0        2.0       -39.0      41.0   
68942       -14.0      35.0      -20.0       49.0        -6.0      55.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
0      1.12132  1.12134  1.12128  1.12132  1.0    0.0       0.0  
1      1.12132  1.12134  1.12124  1.12133  1.0    0.0       0.0  
2      1.12133  1.12142  1.12132  1.12142  0.0    0.0       1.0  
3      1.12142  1.12143  1.12137  1.12142  0.0    1.0       0.0  
4      1.12143  1.12143  1.12122  1.12126  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68938  1.21166  1.21181  1.21147  1.21178  0.0    1.0       0.0  
68939  1.21176  1.21177  1.21153  1.21166  1.0    0.0       0.0  
68940  1.21168  1.21184  1.21164  1.21172  0.0    1.0       0.0  
68941  1.21172  1.21175  1.21134  1.21136  1.0    0.0       0.0  
68942  1.21137  1.21157  1.21102  1.21151  NaN    NaN       NaN  

[68943 rows x 13 columns]

In [17]:
dataset.dropna(inplace=True)

In [18]:
dataset

open_close  open_low  open_high  close_low  close_high  high_low  \
0             0.0       4.0       -2.0        4.0        -2.0       6.0   
1            -1.0       8.0       -2.0        9.0        -1.0      10.0   
2            -9.0       1.0       -9.0       10.0         0.0      10.0   
3             0.0       5.0       -1.0        5.0        -1.0       6.0   
4            17.0      21.0        0.0        4.0       -17.0      21.0   
...           ...       ...        ...        ...         ...       ...   
68937        11.0      12.0      -11.0        1.0       -22.0      23.0   
68938       -12.0      19.0      -15.0       31.0        -3.0      34.0   
68939        10.0      23.0       -1.0       13.0       -11.0      24.0   
68940        -4.0       4.0      -16.0        8.0       -12.0      20.0   
68941        36.0      38.0       -3.0        2.0       -39.0      41.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
0      1.12132  1.12134  1.12128  1.12132  1.0    0.0       0.0  
1      1.12132  1.12134  1.12124  1.12133  1.0    0.0       0.0  
2      1.12133  1.12142  1.12132  1.12142  0.0    0.0       1.0  
3      1.12142  1.12143  1.12137  1.12142  0.0    1.0       0.0  
4      1.12143  1.12143  1.12122  1.12126  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68937  1.21173  1.21184  1.21161  1.21162  1.0    0.0       0.0  
68938  1.21166  1.21181  1.21147  1.21178  0.0    1.0       0.0  
68939  1.21176  1.21177  1.21153  1.21166  1.0    0.0       0.0  
68940  1.21168  1.21184  1.21164  1.21172  0.0    1.0       0.0  
68941  1.21172  1.21175  1.21134  1.21136  1.0    0.0       0.0  

[68942 rows x 13 columns]

In [19]:
#get the absolute value of all the data
absolute_data = abs(dataset)
absolute_data

open_close  open_low  open_high  close_low  close_high  high_low  \
0             0.0       4.0        2.0        4.0         2.0       6.0   
1             1.0       8.0        2.0        9.0         1.0      10.0   
2             9.0       1.0        9.0       10.0         0.0      10.0   
3             0.0       5.0        1.0        5.0         1.0       6.0   
4            17.0      21.0        0.0        4.0        17.0      21.0   
...           ...       ...        ...        ...         ...       ...   
68937        11.0      12.0       11.0        1.0        22.0      23.0   
68938        12.0      19.0       15.0       31.0         3.0      34.0   
68939        10.0      23.0        1.0       13.0        11.0      24.0   
68940         4.0       4.0       16.0        8.0        12.0      20.0   
68941        36.0      38.0        3.0        2.0        39.0      41.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
0      1.12132  1.12134  1.12128  1.12132  1.0    0.0       0.0  
1      1.12132  1.12134  1.12124  1.12133  1.0    0.0       0.0  
2      1.12133  1.12142  1.12132  1.12142  0.0    0.0       1.0  
3      1.12142  1.12143  1.12137  1.12142  0.0    1.0       0.0  
4      1.12143  1.12143  1.12122  1.12126  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68937  1.21173  1.21184  1.21161  1.21162  1.0    0.0       0.0  
68938  1.21166  1.21181  1.21147  1.21178  0.0    1.0       0.0  
68939  1.21176  1.21177  1.21153  1.21166  1.0    0.0       0.0  
68940  1.21168  1.21184  1.21164  1.21172  0.0    1.0       0.0  
68941  1.21172  1.21175  1.21134  1.21136  1.0    0.0       0.0  

[68942 rows x 13 columns]

In [20]:
absolute_data.describe()

open_close      open_low     open_high     close_low    close_high  \
count  68942.000000  68942.000000  68942.000000  68942.000000  68942.000000   
mean      22.869484     21.762786     21.937803     21.906878     21.793711   
std       27.401973     27.831383     27.583420     26.462640     26.558243   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        6.000000      5.000000      5.000000      5.000000      5.000000   
50%       14.000000     13.000000     14.000000     14.000000     14.000000   
75%       30.000000     29.000000     29.000000     29.000000     29.000000   
max      666.000000    585.000000    797.000000    686.000000    618.000000   

           high_low        <OPEN>        <HIGH>         <LOW>       <CLOSE>  \
count  68942.000000  68942.000000  68942.000000  68942.000000  68942.000000   
mean      43.700589      1.135202      1.135422      1.134985      1.135204   
std       36.808987      0.039948      0.039940      0.039955      0.039949   
min        0.000000      1.063870      1.064400      1.063550      1.063890   
25%       20.000000      1.097860      1.098050      1.097602      1.097860   
50%       34.000000      1.127245      1.127480      1.127060      1.127240   
75%       55.000000      1.178000      1.178190      1.177820      1.178000   
max      890.000000      1.211760      1.211840      1.211640      1.211780   

                up*         down*      no_move*  
count  68942.000000  68942.000000  68942.000000  
mean       0.495561      0.477140      0.027298  
std        0.499984      0.499481      0.162952  
min        0.000000      0.000000      0.000000  
25%        0.000000      0.000000      0.000000  
50%        0.000000      0.000000      0.000000  
75%        1.000000      1.000000      0.000000  
max        1.000000      1.000000      1.000000

In [21]:
absolute_data.quantile([.95, .99, .995, .999])

open_close  open_low  open_high  close_low  close_high  high_low  \
0.950        72.0    71.000       71.0       70.0        70.0   110.000   
0.990       128.0   131.000      128.0      123.0       125.0   182.000   
0.995       159.0   163.000      155.0      150.0       155.0   217.000   
0.999       245.0   248.059      257.0      231.0       233.0   323.059   

         <OPEN>    <HIGH>     <LOW>   <CLOSE>  up*  down*  no_move*  
0.950  1.188620  1.188880  1.188380  1.188610  1.0    1.0       0.0  
0.990  1.197000  1.197210  1.196810  1.197006  1.0    1.0       1.0  
0.995  1.204247  1.204558  1.204000  1.204322  1.0    1.0       1.0  
0.999  1.209111  1.209372  1.208961  1.209110  1.0    1.0       1.0

In [22]:
#display frequencies of up, down and no move
dataset[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         34165
0.0  1.0    0.0         32895
     0.0    1.0          1882
dtype: int64

# 95 percentile

# Open_Close Analysis

<li> <b> Anaylyse candles after a prior large drop in prices up to 7.2 pips in a minute open_close </b> </li>

In [23]:
big_drop_oc_95 = dataset[dataset["open_close"] > 72]
big_drop_oc_95

open_close  open_low  open_high  close_low  close_high  high_low  \
49           77.0      82.0        0.0        5.0       -77.0      82.0   
146          80.0      85.0       -6.0        5.0       -86.0      91.0   
326          74.0      77.0       -7.0        3.0       -81.0      84.0   
361          80.0      87.0       -6.0        7.0       -86.0      93.0   
938          85.0     105.0      -19.0       20.0      -104.0     124.0   
...           ...       ...        ...        ...         ...       ...   
68773       131.0     152.0      -45.0       21.0      -176.0     197.0   
68778        91.0      98.0       -6.0        7.0       -97.0     104.0   
68862        98.0     100.0       -5.0        2.0      -103.0     105.0   
68865        81.0      86.0       -2.0        5.0       -83.0      88.0   
68869       119.0     151.0        0.0       32.0      -119.0     151.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
49     1.12134  1.12134  1.12052  1.12057  1.0    0.0       0.0  
146    1.11952  1.11958  1.11867  1.11872  0.0    1.0       0.0  
326    1.11706  1.11713  1.11629  1.11632  0.0    1.0       0.0  
361    1.11426  1.11432  1.11339  1.11346  0.0    1.0       0.0  
938    1.11837  1.11856  1.11732  1.11752  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68773  1.20819  1.20864  1.20667  1.20688  1.0    0.0       0.0  
68778  1.20648  1.20654  1.20550  1.20557  0.0    1.0       0.0  
68862  1.21071  1.21076  1.20971  1.20973  0.0    1.0       0.0  
68865  1.20994  1.20996  1.20908  1.20913  1.0    0.0       0.0  
68869  1.20903  1.20903  1.20752  1.20784  1.0    0.0       0.0  

[1785 rows x 13 columns]

In [24]:
#get the number of up and down movement in price after a big downward movement in price
big_drop_oc_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         984
0.0  1.0    0.0         787
     0.0    1.0          14
dtype: int64

## Result

From the information above we had 984 upward movement in price and 787 downward movement in price total <br/>
and 14 no movement in price <br/>
candles = 984 + 787 + 14 = 1785 <br/>
percentage up = (100 * 984)/(1785) = 55.13%  <br/>
percentage down = (100 * 787)/(1785) = 44.09%  <br/>
percentage no move = (100 * 14)/(1785) = 0.78%   <br/>
so after a 7.2 pips rise in price for a five minute time frame of EURUSD there was a 55.13%  chance that the next candle would be bullish and 44.09% chance of a bearish candle and 0.78% chance of no movement

<li> <b> Anaylyse candles after a prior large drop in prices up to 7.1 pips in a minute open_close </b> </li>

In [25]:
big_rise_oc_95 = dataset[dataset["open_close"] < -71]
big_rise_oc_95

open_close  open_low  open_high  close_low  close_high  high_low  \
143        -174.0       0.0     -193.0      174.0       -19.0     193.0   
1105       -111.0       0.0     -121.0      111.0       -10.0     121.0   
1240        -73.0       4.0      -84.0       77.0       -11.0      88.0   
1841       -199.0       0.0     -214.0      199.0       -15.0     214.0   
1866        -73.0       1.0      -83.0       74.0       -10.0      84.0   
...           ...       ...        ...        ...         ...       ...   
68851       -88.0      17.0      -95.0      105.0        -7.0     112.0   
68858       -86.0       2.0     -123.0       88.0       -37.0     125.0   
68861      -161.0       3.0     -173.0      164.0       -12.0     176.0   
68875       -86.0       1.0      -88.0       87.0        -2.0      89.0   
68909       -75.0       4.0      -78.0       79.0        -3.0      82.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
143    1.11830  1.12023  1.11830  1.12004  0.0    1.0       0.0  
1105   1.11560  1.11681  1.11560  1.11671  0.0    1.0       0.0  
1240   1.11213  1.11297  1.11209  1.11286  0.0    1.0       0.0  
1841   1.10922  1.11136  1.10922  1.11121  0.0    1.0       0.0  
1866   1.11079  1.11162  1.11078  1.11152  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68851  1.20631  1.20726  1.20614  1.20719  0.0    1.0       0.0  
68858  1.20825  1.20948  1.20823  1.20911  1.0    0.0       0.0  
68861  1.20910  1.21083  1.20907  1.21071  0.0    1.0       0.0  
68875  1.20810  1.20898  1.20809  1.20896  0.0    1.0       0.0  
68909  1.20903  1.20981  1.20899  1.20978  1.0    0.0       0.0  

[1695 rows x 13 columns]

In [26]:
big_rise_oc_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
0.0  1.0    0.0         887
1.0  0.0    0.0         791
0.0  0.0    1.0          17
dtype: int64

## Result

From the information above we had 791 upward movement in price and 887 downward movement in price total <br/>
and 17 no move that is open and close are equal
candles = 791 + 887 + 17 = 1695 <br/>
percentage up = (100 * 791)/(1695) = 46.67% <br/>
percentage down = (100 * 887)/(1695) = 52.33% <br/>
percentage no move = (100 * 17)/(1695) = 1.03% <br/>
so after a 7.1 pips rise in price for a five minute time frame of EURUSD there was a 46.67% chance that the next candle would be bullish and 52.33% chance of a bearish candle and 1.03% chance open and close are equal

### Anaylyse candles after a prior large drop in prices up to 7.1 pips in a minute open_low

In [27]:
big_drop_ol_95 = dataset[dataset["open_low"] > 71]
big_drop_ol_95

open_close  open_low  open_high  close_low  close_high  high_low  \
49           77.0      82.0        0.0        5.0       -77.0      82.0   
90           50.0      76.0       -7.0       26.0       -57.0      83.0   
146          80.0      85.0       -6.0        5.0       -86.0      91.0   
152          68.0      73.0       -5.0        5.0       -73.0      78.0   
326          74.0      77.0       -7.0        3.0       -81.0      84.0   
...           ...       ...        ...        ...         ...       ...   
68850        52.0      81.0       -4.0       29.0       -56.0      85.0   
68862        98.0     100.0       -5.0        2.0      -103.0     105.0   
68865        81.0      86.0       -2.0        5.0       -83.0      88.0   
68868        70.0      93.0       -1.0       23.0       -71.0      94.0   
68869       119.0     151.0        0.0       32.0      -119.0     151.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
49     1.12134  1.12134  1.12052  1.12057  1.0    0.0       0.0  
90     1.11947  1.11954  1.11871  1.11897  1.0    0.0       0.0  
146    1.11952  1.11958  1.11867  1.11872  0.0    1.0       0.0  
152    1.11825  1.11830  1.11752  1.11757  0.0    1.0       0.0  
326    1.11706  1.11713  1.11629  1.11632  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68850  1.20683  1.20687  1.20602  1.20631  1.0    0.0       0.0  
68862  1.21071  1.21076  1.20971  1.20973  0.0    1.0       0.0  
68865  1.20994  1.20996  1.20908  1.20913  1.0    0.0       0.0  
68868  1.20973  1.20974  1.20880  1.20903  0.0    1.0       0.0  
68869  1.20903  1.20903  1.20752  1.20784  1.0    0.0       0.0  

[3436 rows x 13 columns]

In [28]:
#get the number of up and down movement in price after a big downward movement in price
big_drop_ol_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         1798
0.0  1.0    0.0         1609
     0.0    1.0           29
dtype: int64

## Result

From the information above we had 1798 upward movement in price and 1609 downward movement in price total <br/>
and 29 no move that is open and close are equal
candles = 1798 + 1609 + 29 = 3436  <br/>
percentage up = (100 * 1798)/(3436) = 52.33% <br/>
percentage down = (100 * 1609)/(3436) = 46.83% <br/>
percentage no move = (100 * 29)/(3436) = 0.84% <br/>
so after a 7.1 pips drop in price for a five minute time frame of EURUSD there was a 52.33% chance that the next candle would be bullish and 46.83% chance of a bearish candle and 0.84% chance open and close are equal

### Anaylyse candles after a prior large drop in prices up to 7.1 pips in a minute open_low

In [29]:
big_rise_oh_95 = dataset[dataset["open_high"] < -71]
big_rise_oh_95

open_close  open_low  open_high  close_low  close_high  high_low  \
143        -174.0       0.0     -193.0      174.0       -19.0     193.0   
420         -37.0       0.0     -149.0       37.0      -112.0     149.0   
1105       -111.0       0.0     -121.0      111.0       -10.0     121.0   
1136        -56.0       3.0      -74.0       59.0       -18.0      77.0   
1240        -73.0       4.0      -84.0       77.0       -11.0      88.0   
...           ...       ...        ...        ...         ...       ...   
68864       -52.0       1.0      -96.0       53.0       -44.0      97.0   
68866       -34.0       1.0      -77.0       35.0       -43.0      78.0   
68875       -86.0       1.0      -88.0       87.0        -2.0      89.0   
68877       -42.0       1.0      -74.0       43.0       -32.0      75.0   
68909       -75.0       4.0      -78.0       79.0        -3.0      82.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
143    1.11830  1.12023  1.11830  1.12004  0.0    1.0       0.0  
420    1.11522  1.11671  1.11522  1.11559  1.0    0.0       0.0  
1105   1.11560  1.11681  1.11560  1.11671  0.0    1.0       0.0  
1136   1.11533  1.11607  1.11530  1.11589  1.0    0.0       0.0  
1240   1.11213  1.11297  1.11209  1.11286  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68864  1.20943  1.21039  1.20942  1.20995  0.0    1.0       0.0  
68866  1.20913  1.20990  1.20912  1.20947  1.0    0.0       0.0  
68875  1.20810  1.20898  1.20809  1.20896  0.0    1.0       0.0  
68877  1.20885  1.20959  1.20884  1.20927  0.0    1.0       0.0  
68909  1.20903  1.20981  1.20899  1.20978  1.0    0.0       0.0  

[3386 rows x 13 columns]

In [30]:
#get the number of up and down movement in price after a big upward movement in price
big_rise_oh_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
0.0  1.0    0.0         1719
1.0  0.0    0.0         1641
0.0  0.0    1.0           26
dtype: int64

## Result

From the information above we had 1641 upward movement in price and 1719 downward movement in price total <br/>
and 26 no move that is open and close are equal
candles = 1641 + 1719 + 26 = 3386 <br/>
percentage up = (100 * 1641)/(3386) = 48.46% <br/>
percentage down = (100 * 1719)/(3386) = 50.77% <br/>
percentage no move = (100 * 26)/(3386) = 0.77% <br/>
so after a 7.1 pips drop in price for a five minute time frame of EURUSD there was a 48.46% chance that the next candle would be bullish and 50.77% chance of a bearish candle and 0.77% chance open and close are equal

# Close_Low Analysis

<b><li> Anaylyse candles after a prior large move in prices up to 7.0 pips in a five minute close_low that is bearish </li></b>

In [31]:
big_move_cl_bear_95 = dataset[dataset["close_low"] > 70]
big_move_cl_bear_95 = big_move_cl_bear_95[big_move_cl_bear_95["open_close"] > 0]
big_move_cl_bear_95

open_close  open_low  open_high  close_low  close_high  high_low  \
1201         88.0     201.0        0.0      113.0       -88.0     201.0   
10575         9.0     144.0      -12.0      135.0       -21.0     156.0   
11889         9.0      86.0       -2.0       77.0       -11.0      88.0   
12144        35.0     105.0      -10.0       70.0       -45.0     115.0   
12177        10.0     107.0       -7.0       97.0       -17.0     114.0   
...           ...       ...        ...        ...         ...       ...   
65200        24.0     108.0       -2.0       84.0       -26.0     110.0   
66210         5.0      82.0       -4.0       77.0        -9.0      86.0   
68283        19.0     100.0      -74.0       81.0       -93.0     174.0   
68293       164.0     248.0       -1.0       84.0      -165.0     249.0   
68585        32.0     105.0       -2.0       73.0       -34.0     107.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
1201   1.11515  1.11515  1.11314  1.11427  0.0    1.0       0.0  
10575  1.08250  1.08262  1.08106  1.08241  1.0    0.0       0.0  
11889  1.10133  1.10135  1.10047  1.10124  0.0    1.0       0.0  
12144  1.10886  1.10896  1.10781  1.10851  0.0    1.0       0.0  
12177  1.10998  1.11005  1.10891  1.10988  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
65200  1.18378  1.18380  1.18270  1.18354  1.0    0.0       0.0  
66210  1.18245  1.18249  1.18163  1.18240  0.0    1.0       0.0  
68283  1.19957  1.20031  1.19857  1.19938  1.0    0.0       0.0  
68293  1.19800  1.19801  1.19552  1.19636  0.0    1.0       0.0  
68585  1.20481  1.20483  1.20376  1.20449  1.0    0.0       0.0  

[239 rows x 13 columns]

In [32]:
big_move_cl_bear_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         120
0.0  1.0    0.0         117
     0.0    1.0           2
dtype: int64

## Result

From the information above we had 120 upward movement in price and 117 downward movement in price total <br/>
and 2 no move that is open and close are equal
candles = 120 + 117 + 27 = 239 <br/>
percentage up = (100 * 120)/(239) = 50.21% <br/>
percentage down = (100 * 117)/(239) = 48.95% <br/>
percentage no move = (100 * 2)/(239) = 0.84% <br/>
so after a 7.0 pips drop in price for a one minute time frame of EURUSD there was a 50.21% chance that the next candle would be bullish and 48.95% chance of a bearish candle and 0.84% chance open and close are equal

<b><li> Anaylyse candles after a prior large move in prices up to 7.0 pips in a minute close_low that is bullish </li></b>

In [33]:
big_move_cl_bull_95 = dataset[dataset["close_low"] > 70]
big_move_cl_bull_95 = big_move_cl_bull_95[big_move_cl_bull_95["open_close"] < 0]
big_move_cl_bull_95

open_close  open_low  open_high  close_low  close_high  high_low  \
143        -174.0       0.0     -193.0      174.0       -19.0     193.0   
430         -49.0      34.0      -62.0       83.0       -13.0      96.0   
995         -35.0      42.0      -37.0       77.0        -2.0      79.0   
1105       -111.0       0.0     -121.0      111.0       -10.0     121.0   
1240        -73.0       4.0      -84.0       77.0       -11.0      88.0   
...           ...       ...        ...        ...         ...       ...   
68854       -60.0      11.0      -80.0       71.0       -20.0      91.0   
68858       -86.0       2.0     -123.0       88.0       -37.0     125.0   
68861      -161.0       3.0     -173.0      164.0       -12.0     176.0   
68875       -86.0       1.0      -88.0       87.0        -2.0      89.0   
68909       -75.0       4.0      -78.0       79.0        -3.0      82.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
143    1.11830  1.12023  1.11830  1.12004  0.0    1.0       0.0  
430    1.11594  1.11656  1.11560  1.11643  0.0    1.0       0.0  
995    1.11457  1.11494  1.11415  1.11492  0.0    1.0       0.0  
1105   1.11560  1.11681  1.11560  1.11671  0.0    1.0       0.0  
1240   1.11213  1.11297  1.11209  1.11286  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68854  1.20747  1.20827  1.20736  1.20807  0.0    1.0       0.0  
68858  1.20825  1.20948  1.20823  1.20911  1.0    0.0       0.0  
68861  1.20910  1.21083  1.20907  1.21071  0.0    1.0       0.0  
68875  1.20810  1.20898  1.20809  1.20896  0.0    1.0       0.0  
68909  1.20903  1.20981  1.20899  1.20978  1.0    0.0       0.0  

[3163 rows x 13 columns]

In [34]:
big_move_cl_bull_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
0.0  1.0    0.0         1618
1.0  0.0    0.0         1517
0.0  0.0    1.0           28
dtype: int64

## Result

From the information above we had upward 1517 movement in price and  1618 downward movement in price total <br/>
and 28 no move that is open and close are equal
candles =  1517 + 1618 + 28 = 3163<br/>
percentage up = (100 * 1517)/(3163) = 47.96% <br/>
percentage down = (100 * 1618)/(3163) = 51.15% <br/>
percentage no move = (100 * 28)/(3163) = 0.89% <br/>
so after a 7.0 pips drop in price for a five minute time frame of EURUSD there was a 47.96% chance that the next candle would be bearish and 51.15% chance of a bullish candle and 0.89% chance open and close are equal

# Close_High Analysis

<b> <li> Anaylyse candles after a prior large move in prices up to 7.0 pips in a minute close_high when the candle is bearish </li> </b>

In [35]:
big_drop_ch_95 = dataset[dataset["close_high"] < -70]
big_drop_ch_95 = big_drop_ch_95[big_drop_ch_95["open_close"] > 0]
big_drop_ch_95

open_close  open_low  open_high  close_low  close_high  high_low  \
49           77.0      82.0        0.0        5.0       -77.0      82.0   
146          80.0      85.0       -6.0        5.0       -86.0      91.0   
152          68.0      73.0       -5.0        5.0       -73.0      78.0   
326          74.0      77.0       -7.0        3.0       -81.0      84.0   
361          80.0      87.0       -6.0        7.0       -86.0      93.0   
...           ...       ...        ...        ...         ...       ...   
68855        51.0      54.0      -22.0        3.0       -73.0      76.0   
68862        98.0     100.0       -5.0        2.0      -103.0     105.0   
68865        81.0      86.0       -2.0        5.0       -83.0      88.0   
68868        70.0      93.0       -1.0       23.0       -71.0      94.0   
68869       119.0     151.0        0.0       32.0      -119.0     151.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
49     1.12134  1.12134  1.12052  1.12057  1.0    0.0       0.0  
146    1.11952  1.11958  1.11867  1.11872  0.0    1.0       0.0  
152    1.11825  1.11830  1.11752  1.11757  0.0    1.0       0.0  
326    1.11706  1.11713  1.11629  1.11632  0.0    1.0       0.0  
361    1.11426  1.11432  1.11339  1.11346  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68855  1.20807  1.20829  1.20753  1.20756  1.0    0.0       0.0  
68862  1.21071  1.21076  1.20971  1.20973  0.0    1.0       0.0  
68865  1.20994  1.20996  1.20908  1.20913  1.0    0.0       0.0  
68868  1.20973  1.20974  1.20880  1.20903  0.0    1.0       0.0  
68869  1.20903  1.20903  1.20752  1.20784  1.0    0.0       0.0  

[3129 rows x 13 columns]

In [36]:
big_drop_ch_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         1680
0.0  1.0    0.0         1419
     0.0    1.0           30
dtype: int64

## Result

From the information above we had 1680 upward movement in price and 1419 downward movement in price total <br/>
and 30 no move that is open and close are equal
candles = 1680 + 1419 + 30 = 3129 <br/>
percentage up = (100 * 1680)/(3129) = 53.69% <br/>
percentage down = (100 * 1419)/(3129) = 45.35% <br/>
percentage no move = (100 * 30)/(3129) = 0.96% <br/>
so after a 7.0 pips drop in price for a one minute time frame of EURUSD there was a 53.69% chance that the next candle would be bullish and 45.35% chance of a bearish candle and 0.96% chance open and close are equal

### Anaylyse candles after a prior large move in prices up to 7.0 pips in a minute close_high when the candle is bullish

In [37]:
big_rise_ch_95 = dataset[dataset["close_high"] < -70]
big_rise_ch_95 = big_rise_ch_95[big_rise_ch_95["open_close"] < 0]
big_rise_ch_95

open_close  open_low  open_high  close_low  close_high  high_low  \
420         -37.0       0.0     -149.0       37.0      -112.0     149.0   
6188        -17.0      10.0      -98.0       27.0       -81.0     108.0   
9574        -34.0      14.0     -129.0       48.0       -95.0     143.0   
11657        -3.0      10.0     -116.0       13.0      -113.0     126.0   
12229      -106.0      18.0     -178.0      124.0       -72.0     196.0   
...           ...       ...        ...        ...         ...       ...   
63929       -50.0       1.0     -197.0       51.0      -147.0     198.0   
63930        -4.0      72.0      -96.0       76.0       -92.0     168.0   
63931       -52.0      22.0     -142.0       74.0       -90.0     164.0   
64253       -30.0      19.0     -105.0       49.0       -75.0     124.0   
67141       -34.0      19.0     -106.0       53.0       -72.0     125.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
420    1.11522  1.11671  1.11522  1.11559  1.0    0.0       0.0  
6188   1.10805  1.10903  1.10795  1.10822  1.0    0.0       0.0  
9574   1.08239  1.08368  1.08225  1.08273  1.0    0.0       0.0  
11657  1.09756  1.09872  1.09746  1.09759  1.0    0.0       0.0  
12229  1.11203  1.11381  1.11185  1.11309  1.0    0.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
63929  1.18726  1.18923  1.18725  1.18776  1.0    0.0       0.0  
63930  1.18776  1.18872  1.18704  1.18780  1.0    0.0       0.0  
63931  1.18780  1.18922  1.18758  1.18832  1.0    0.0       0.0  
64253  1.18237  1.18342  1.18218  1.18267  0.0    1.0       0.0  
67141  1.18718  1.18824  1.18699  1.18752  0.0    1.0       0.0  

[277 rows x 13 columns]

In [38]:
big_rise_ch_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         150
0.0  1.0    0.0         126
     0.0    1.0           1
dtype: int64

## Result

From the information above we had 150 upward movement in price and 126 downward movement in price total <br/>
and 1 no move that is open and close are equal
candles = 150 + 126 + 1 = 271 <br/>
percentage up = (100 * 150)/(277) = 54.15% <br/>
percentage down = (100 * 129)/(277) = 46.57% <br/>
percentage no move = (100 * 1)/(277) = 0.36% <br/>
so after a 7.0 pips move in price for a one minute time frame of EURUSD there was a 54.15% chance that the next candle would be bullish and 46.57% chance of a bearish candle and 0.36% chance open and close are equal

### Anaylyse candles after a prior large drop in prices up to 11 pips in a minute high_low

In [39]:
big_drop_hl_95 = dataset[dataset["high_low"] > 110]
big_drop_hl_95 = big_drop_hl_95[big_drop_hl_95["open_close"] > 0]
big_drop_hl_95

open_close  open_low  open_high  close_low  close_high  high_low  \
938          85.0     105.0      -19.0       20.0      -104.0     124.0   
1201         88.0     201.0        0.0      113.0       -88.0     201.0   
1202         66.0     117.0      -12.0       51.0       -78.0     129.0   
4278        130.0     152.0       -3.0       22.0      -133.0     155.0   
4431        143.0     151.0       -3.0        8.0      -146.0     154.0   
...           ...       ...        ...        ...         ...       ...   
68297        97.0     112.0       -2.0       15.0       -99.0     114.0   
68581        53.0     107.0      -51.0       54.0      -104.0     158.0   
68587       115.0     116.0       -1.0        1.0      -116.0     117.0   
68773       131.0     152.0      -45.0       21.0      -176.0     197.0   
68869       119.0     151.0        0.0       32.0      -119.0     151.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
938    1.11837  1.11856  1.11732  1.11752  0.0    1.0       0.0  
1201   1.11515  1.11515  1.11314  1.11427  0.0    1.0       0.0  
1202   1.11427  1.11439  1.11310  1.11361  1.0    0.0       0.0  
4278   1.10948  1.10951  1.10796  1.10818  1.0    0.0       0.0  
4431   1.11084  1.11087  1.10933  1.10941  0.0    1.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68297  1.19567  1.19569  1.19455  1.19470  1.0    0.0       0.0  
68581  1.20368  1.20419  1.20261  1.20315  1.0    0.0       0.0  
68587  1.20537  1.20538  1.20421  1.20422  1.0    0.0       0.0  
68773  1.20819  1.20864  1.20667  1.20688  1.0    0.0       0.0  
68869  1.20903  1.20903  1.20752  1.20784  1.0    0.0       0.0  

[1721 rows x 13 columns]

In [40]:
big_drop_hl_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         912
0.0  1.0    0.0         797
     0.0    1.0          12
dtype: int64

## Result

From the information above we had 912 upward movement in price and 797 downward movement in price total <br/>
and 12 no move that is open and close are equal
candles = 912 + 797 + 12 = 1721  <br/>
percentage up = (100 * 912)/(1721) = 52.99% <br/>
percentage down = (100 * 797)/(1721) = 46.31% <br/>
percentage no move = (100 * 12)/(1721) = 0.70% <br/>
so after a 11 pips drop in price for a one minute time frame of EURUSD there was a 52.99% chance that the next candle would be bullish and 46.31% chance of a bearish candle and 0.70% chance open and close are equal

### Anaylyse candles after a prior large rise in prices up to 11.0 pips in a minute high_low

In [41]:
big_rise_hl_95 = dataset[dataset["high_low"] > 46]
big_rise_hl_95 = big_rise_hl_95[big_rise_hl_95["open_close"] < 0]
big_rise_hl_95

open_close  open_low  open_high  close_low  close_high  high_low  \
48          -37.0      11.0      -38.0       48.0        -1.0      49.0   
83          -40.0       0.0      -53.0       40.0       -13.0      53.0   
84          -30.0      14.0      -33.0       44.0        -3.0      47.0   
96          -27.0      17.0      -32.0       44.0        -5.0      49.0   
137          -2.0      28.0      -20.0       30.0       -18.0      48.0   
...           ...       ...        ...        ...         ...       ...   
68910       -41.0       3.0      -46.0       44.0        -5.0      49.0   
68912       -21.0       6.0      -40.0       27.0       -19.0      46.0   
68927       -29.0      11.0      -38.0       40.0        -9.0      49.0   
68934       -44.0       3.0      -65.0       47.0       -21.0      68.0   
68935       -38.0       1.0      -49.0       39.0       -11.0      50.0   

        <OPEN>   <HIGH>    <LOW>  <CLOSE>  up*  down*  no_move*  
48     1.12098  1.12136  1.12087  1.12135  0.0    1.0       0.0  
83     1.11942  1.11995  1.11942  1.11982  1.0    0.0       0.0  
84     1.11981  1.12014  1.11967  1.12011  1.0    0.0       0.0  
96     1.11847  1.11879  1.11830  1.11874  1.0    0.0       0.0  
137    1.11737  1.11757  1.11709  1.11739  1.0    0.0       0.0  
...        ...      ...      ...      ...  ...    ...       ...  
68910  1.20977  1.21023  1.20974  1.21018  0.0    1.0       0.0  
68912  1.20987  1.21027  1.20981  1.21008  0.0    1.0       0.0  
68927  1.21008  1.21046  1.20997  1.21037  0.0    1.0       0.0  
68934  1.21078  1.21143  1.21075  1.21122  1.0    0.0       0.0  
68935  1.21122  1.21171  1.21121  1.21160  1.0    0.0       0.0  

[11872 rows x 13 columns]

In [46]:
big_rise_hl_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
0.0  1.0    0.0         5956
1.0  0.0    0.0         5793
0.0  0.0    1.0          123
dtype: int64

## Result

From the information above we had 5793 upward movement in price and 5956 downward movement in price total <br/>
and 123 no move that is open and close are equal
candles = 5793 + 5956 + 123 = 11872 <br/>
percentage up = (100 * 5793)/(11872) = 48.8% <br/>
percentage down = (100 * 5956)/(11872) = 50.18% <br/>
percentage no move = (100 * 123)/(11872) = 1.04% <br/>
so after a 4.6 pips drop in price for a five minute time frame of EURUSD there was a 48.8% chance that the next candle would be bullish and 50.18% chance of a bearish candle and 1.95% chance open and close are equal